nwbzarr/nwbzarr.py

```diff
@@ -102,43 +73,51 @@ def __init__(self, nwbfile: str = None, nwbgroup: str = None, nwbfile_mode: str
         # dictionary to hold addresses of hdf5 objects in file
         self._address_dict = {}
 
-        self.changed_dsets = {}
         # create zarr format hierarchy for datasets and attributes compatible with NWB file,
-        # dataset contents are not copied, unless allow_changing_string_type is False or
-        # nwbfile_mode == 'r', and NWBFile contains variable-length strings
+        # dataset contents are not copied, unless it contains variable-length strings
+
+        self.nwb_zgroup = zarr.open_group(self.store, mode=self.store_mode, path=self.store_path)
+        if self.store is None:
+            self.store = self.nwb_zgroup.store
 
         # FileChunkStore requires uri
         if isinstance(nwbfile, str):
             self.uri = nwbfile
         else:
             self.uri = nwbfile.path
 
-        # Create zarr hierarchy
-        self.nwb_zgroup = zarr.open_group(self.store, mode=self.store_mode, path=self.store_path)
-        if self.store is None:
-            self.store = self.nwb_zgroup.store
-        self.create_zarr_hierarchy(self.group, self.nwb_zgroup)
+        # Access NWB file and create zarr hierarchy
+        if nwbgroup and not isinstance(nwbgroup, str):
+            raise TypeError(f"Expected str for nwbgroup, recieved {type(nwbgroup)}")
+        self.nwbgroup = nwbgroup
+        self.nwbfile = nwbfile
+        if self.store_mode != 'r':
+            self.file = h5py.File(self.nwbfile, mode=self.nwbfile_mode)
+            self.group = self.file[self.nwbgroup] if self.nwbgroup else self.file
+            self.create_zarr_hierarchy(self.group, self.nwb_zgroup)
+            self.file.close()
         if isinstance(self.nwbfile, str):
-            fsspec_file = fsspec.open(self.nwbfile, mode='rb')
-            chunk_store = FileChunkStore(self.store, chunk_source=fsspec_file.open())
+            self.chunkstore_file = fsspec.open(self.nwbfile, mode='rb')
+            self.chunk_store = FileChunkStore(self.store, chunk_source=self.chunkstore_file.open())
         else:
-            chunk_store = FileChunkStore(self.store, chunk_source=self.nwbfile)
-        if LRU is True and not isinstance(chunk_store, zarr.LRUStoreCache):
-            chunk_store = zarr.LRUStoreCache(chunk_store, max_size=self.LRU_max_size)
+            self.chunk_store = FileChunkStore(self.store, chunk_source=self.nwbfile)
```

Comment:
If `nwbfile` can be of type `File`, indicate that in the docstring

Reply:
Class is renamed in 9f277a8. Docstring is updated as well.

Reply:
Added an updated README in PR #2